tools/GenerateRollUpModule.ps1

```diff
@@ -69,17 +69,24 @@ elseif ($VersionState.Equals([VersionState]::Valid) -or $VersionState.Equals([Ve
     if (-not (Test-Path $GraphModuleLocation)) {
         New-Item -Path $GraphModuleLocation -Type Directory
     }
+    $AllowPreRelease = $false
+    if($ModulePreviewNumber -ge 0) {
+        $AllowPreRelease = $true
+    }
 
     # Add auth module as a dependency.
-    $ExistingAuthModule = Find-Module "Microsoft.Graph.Authentication"
-    Install-Module $ExistingAuthModule.Name -Repository $RepositoryName -AllowPrerelease -Force
+    Find-Module "Microsoft.Graph.Authentication" -Repository $RepositoryName -AllowPreRelease:$AllowPreRelease
+    $ExistingAuthModule = Find-Module "Microsoft.Graph.Authentication" -Repository $RepositoryName -AllowPreRelease:$AllowPreRelease
+    Install-Module $ExistingAuthModule.Name -Repository $RepositoryName -Force -AllowClobber -AllowPreRelease:$AllowPreRelease
     $RequiredGraphModules += @{ ModuleName = $ExistingAuthModule.Name ; ModuleVersion = $ExistingAuthModule.Version }
 
     foreach ($RequiredModule in $ModuleMapping.Keys) {
         # Install module locally in order to specify it as a dependency of the roll-up module down the generation pipeline.
         # https://stackoverflow.com/questions/46216038/how-do-i-define-requiredmodules-in-a-powershell-module-manifest-psd1.
-        $ExistingWorkloadModule = Find-Module "$ModulePrefix.$RequiredModule"
-        Install-Module $ExistingWorkloadModule.Name -Repository $RepositoryName -AllowPrerelease -Force
+        $ExistingWorkloadModule = Find-Module "$ModulePrefix.$RequiredModule" -Repository $RepositoryName -AllowPreRelease:$AllowPreRelease
+        Write-Warning "Installing $ModulePrefix.$RequiredModule Version: $ExistingWorkloadModule.Version"
+        Write-Warning $ExistingWorkloadModule
```

Comment:
No need to log the module object. We have already logged its name and version in the line before.

Reply:
I added the logging statements to see what the pipeline was doing. I will remove them when successful